lib/iris/common/metadata.py

```diff
@@ -227,6 +227,15 @@ def __ne__(self, other):
 
         return result
 
+    def __str__(self):
+        field_strings = []
+        for field in self._fields:
+            value = getattr(self, field)
+            if value is not None:
```

Comment:
@rcomer I think that we could be even more liberal here and ignore empty strings perhaps, but definitely and empty `dict` for the `attributes` member.

Reply:
It took me longer than I care to admit to figure out the logic for this 😆

Reply:
In fact it's still not right - I appear to have lost my boolean (False) elements somehow...